cli/js/net.ts

```diff
@@ -109,41 +110,28 @@ export class ConnImpl implements Conn {
 }
 
 export class ListenerImpl implements Listener {
-  constructor(
-    readonly rid: number,
-    readonly addr: Addr,
-    private closing: boolean = false
-  ) {}
+  constructor(readonly rid: number, readonly addr: Addr) {}
 
   async accept(): Promise<Conn> {
     const res = await sendAsync("op_accept", { rid: this.rid });
     return new ConnImpl(res.rid, res.remoteAddr, res.localAddr);
   }
 
   close(): void {
-    this.closing = true;
     close(this.rid);
   }
 
-  async next(): Promise<IteratorResult<Conn>> {
-    if (this.closing) {
-      return { value: undefined, done: true };
-    }
-    return await this.accept()
-      .then(value => ({ value, done: false }))
-      .catch(e => {
-        // It wouldn't be correct to simply check this.closing here.
-        // TODO: Get a proper error kind for this case, don't check the message.
-        // The current error kind is Other.
-        if (e.message == "Listener has been closed") {
-          return { value: undefined, done: true };
+  async *[Symbol.asyncIterator](): AsyncIterator<Conn> {
+    while (true) {
+      try {
+        yield await this.accept();
+      } catch (error) {
+        if (error instanceof errors.BadResource) {
+          break;
```

Comment:
Why does BadResource need to be handled specially?

Reply:
If the error is that the `Listener` is closed, we just want to break. Before this was caught as an `Other` error.